floodsub.go

```diff
@@ -16,7 +16,11 @@ import (
 	timecache "github.com/whyrusleeping/timecache"
 )
 
-const ID = protocol.ID("/floodsub/1.0.0")
+const (
+	ID                     = protocol.ID("/floodsub/1.0.0")
+	maxConcurrency         = 10
```

Comment:
Let's make this configurable too?

Reply:
it doesn't have to be configurable per subscription -- I just don't like hardcoding a low value.
But up to you.

Reply:
Where else would you make it configurable? As a parameter to NewFloodSub? Then we had a global value for all subscriptions, and we would need to hard code a value in go-ipfs. I don't think that is a good option either.

I agree that 10 might not be the best value, so I sat down and did some simulations. They are quite rough, but I think we can still use them to inform our decision. You can find the code and "results" at https://gist.github.com/keks/a3bd10097c9c1d20ef5a2de3b5f794e3. Based on that, I'd suggest a value of 128. That seems quite high, but I believe we might drop too many packets otherwise.

The simulation assumes message validation takes as long as its timeout, so it's a worst-case analysis.

What do you think @whyrusleeping @vyzo?

Reply:
It can be either a parameter in `NewFloodSub` or just make it a variable.
Either way, if we make the validation timeout configurable, we should make the max queue size configurable as well.

Reply:
Yeah, maybe we make this an option on `NewFloodsub`. When adding 'options' to things, i really like this approach: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
I'm not convinced a global setting is helpful here, but I'm okay going with it. And sure, functional options are nice! That's how I implemented other options in this PR as well.